### PR TITLE
[Inference API] Add Google Vertex AI Rerank support

### PR DESCRIPTION
**Pull request by timgrein**
Adds support for Google Vertex AI reranking through [Google Vertex AI Agent Builder Rank and Rerank documents](https://cloud.google.com/generative-ai-app-builder/docs/ranking).

Creating a `rerank` inference endpoint using `googlevertexai` as service provider (they always use the same default model):

```
PUT {{ES_HOST}}/_inference/rerank/google_vertex_ai_reranking

{
    "service": "googlevertexai",
    "service_settings": {
        "service_account_json": "<service_account_json>",
        "project_id": "<project_id>"
    }
}
```

Reranking a set of documents
```
POST {{ES_HOST}}/_inference/rerank/google_vertex_ai_reranking

{
  "input": ["like", "luke skywalker", "leia", "chewy","r2d2", "star", "wars", "harry potter"],
  "query": "harry potter"
}
```

### CODE REVIEW

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/googlevertexai/GoogleVertexAiRerankActionTests.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @After
+    public void shutdown() throws IOException {
+        clientManager.close();
+        terminate(threadPool);
+        webServer.close();
+    }
+
+    // Successful case tested via end-to-end notebook tests in AppEx repo
```
Comment (jonathan-buttner) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/googlevertexai/GoogleVertexAiRerankActionTests.java:63]: Just for my own edification, could you link me to the tests?

**Review by jonathan-buttner — APPROVED** [review]
Great work, just had a few questions.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/GoogleVertexAiEmbeddingsRequestManager.java [hunk 2]
```diff
@@ -41,11 +41,19 @@ private static ResponseHandler createEmbeddingsHandler() {
     private final Truncator truncator;
 
     public GoogleVertexAiEmbeddingsRequestManager(GoogleVertexAiEmbeddingsModel model, Truncator truncator, ThreadPool threadPool) {
-        super(threadPool, model);
+        super(threadPool, model, RateLimitGrouping.of(model));
         this.model = Objects.requireNonNull(model);
         this.truncator = Objects.requireNonNull(truncator);
     }
 
+    record RateLimitGrouping(int modelIdHash) {
```
[the remark this answers is not in the document]
Comment (timgrein) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/GoogleVertexAiEmbeddingsRequestManager.java:49]: I can check that for embeddings and create a follow-up PR 👍

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/googlevertexai/GoogleVertexAiRerankActionTests.java (hunk 1 above)
Comment (timgrein) [reply x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/googlevertexai/GoogleVertexAiRerankActionTests.java:63]: Will send you on Slack :)

**Pull request merged**